[scarletblanks] iP

ScarletBlanks opened the pull request:
Till


Till frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/ScarletBlanks/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!



Features:
- [X] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)




If you Java programmer, you can use it to practice Java too. Here's the main method:

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ScarletBlanks committed:
Level-0 Name Change

ScarletBlanks committed:
Improve the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye

ScarletBlanks committed:
Improve the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye.

ScarletBlanks committed:
Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

ScarletBlanks committed:
Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

ScarletBlanks committed:
fixed a small error where the number wasn't increasing

ScarletBlanks committed:
fixed a small error where the number wasn't increasing

ScarletBlanks committed:
Line issues fixed

ScarletBlanks committed:
Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

ScarletBlanks committed:
Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

ScarletBlanks committed:
Level 3. Mark as Done
Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.
With OOP

ScarletBlanks committed:
Try to push to lv3

ScarletBlanks committed:
Level 4. ToDos, Events, Deadlines
Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

ScarletBlanks committed:
Level-4 fixes

ScarletBlanks committed:
Level 5. Handle Errors
Teach the chatbot to deal with errors such as incorrect inputs entered by the user.

ScarletBlanks committed:
Level 6. Delete
Add support for deleting tasks from the list.

ScarletBlanks committed:
A-TextUiTesting

ScarletBlanks committed:
Level-7 Completed. Save feature added, Load feature added, Modified the Task/Deadline/Todo/Event files to achieve said feature. Duke file also updated to enable loading and saving.

ScarletBlanks committed:
Account for  the folder-not-exists error by using mkdir

ScarletBlanks committed:
Created consistency by filtering text using " /from " and " /by " such that we will be able to use "/" to filter date later on with no problem.

ScarletBlanks committed:
Overhaulled the date/time system for Deadlines, Duke now recognises Dates and Times for Deadlines and can convert them to different formats.

ScarletBlanks committed:
Merge branch 'branch-Level-8'

ScarletBlanks committed:
Implemented the UI class to deal with printing for user interactions.

ScarletBlanks committed:
Storage class added to manage the loading and saving of tasks

ScarletBlanks committed:
Parser added : works as a 2-in-1 parse and command : deals with making sense of the user command

ScarletBlanks committed:
Added TaskList, contains the task list e.g., it has operations to add/delete tasks in the list, also modified every other java file to accomodate for this change i.e change from ArrayList<Task> myList to using TaskList tasks to handle adds/deletes

ScarletBlanks committed:
A-Packages, put everything into duke package

ScarletBlanks committed:
duke package

ScarletBlanks committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ScarletBlanks committed:
Set up gradle so that you can build and run Duke using gradle.

ScarletBlanks committed:
Added testcases for JUNIT implementation

ScarletBlanks committed:
A-Jar: Create a JAR File

ScarletBlanks committed:
added the find feature

ScarletBlanks committed:
Added JavaDocs to Parser and Ui

ScarletBlanks committed:
Merge branch 'branch-Level9'

ScarletBlanks committed:
Merge tag 'A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Ui.java

ScarletBlanks committed:
javadoc added

ScarletBlanks committed:
Fixed the bug where the next line after "find" would not work because it didn't break; properly

ScarletBlanks committed:
Checkstyle Commits

ScarletBlanks committed:
Checkstyle Changes

ScarletBlanks committed:
Added GUI

ScarletBlanks committed:
GUI added

xxiaoweii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,143 @@
+            break;
+
+        case "mark":
+            int number = myScanner.nextInt();
```
The naming can be more consistent to improve clarity (number is used here but numero is used at the later part)

xxiaoweii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,143 @@
+            Ui.unrecognisedCommand();
+            break;
+
+        }
```
you might want to close the scanner at the end

xxiaoweii commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,82 @@
+                nextLine = bufferedReader.readLine();
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
```
i think it's better to throw different types of error instead of printing the error messages

xxiaoweii commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,174 @@
+     *
+     * @param item the item that has been unmarked
+     */
+    public void taskUndone(Task item) {
```
The naming of this method can be more specific to suggest that it is a method to print the messages. Same for taskAdd and taskDelete

xxiaoweii reviewed (commented):
Overall good effort! Just some minor changes

ScarletBlanks committed:
improved coding standard

ScarletBlanks committed:
feat: Add ParserGUI and UiGUI, update build.gradle

- Added a new ParserGUI class for GUI parsing.
- Added a new UiGUI class for GUI user interface.
- Updated build.gradle to run the application using Launcher for GUI.
- Note: Full conversion to a complete UI is still a work in progress.

ScarletBlanks committed:
Improved code quality

ScarletBlanks committed:
docs: Update DaDuke.png, DaUser.png, and Java files for GUI functionality

- Update DaDuke.png with the latest changes.
- Update DaUser.png with the latest changes.
- Modify UiGUI, ParserGUI, and Duke.java to enable full GUI functionality.
- Clean up text for improved readability in GUI format.

ScarletBlanks committed:
Adding more assertion updates to replace this one in a future update.

ScarletBlanks committed:
Add EventTest.java and TaskTest.java for unit testing

ScarletBlanks committed:
Made Duke greet the user on entry for the GUI

ScarletBlanks committed:
Implemented B-Reminders

ScarletBlanks committed:
test

ScarletBlanks committed:
Merge pull request #2 from ScarletBlanks/branch-PR-test

test

ScarletBlanks committed:
Awarding credit for JavaFX tutorial

ScarletBlanks committed:
Updated the README and added a picture of the User Interface.

ScarletBlanks committed:
Update README.md

ScarletBlanks committed:
Added Javadocs for at least half of the classes

ScarletBlanks committed:
Merge branch 'master' of https://github.com/ScarletBlanks/ip

ScarletBlanks committed:
Handle errors using exception

ScarletBlanks committed:
Handle Errors using exceptions

ScarletBlanks committed:
Update README.md

ScarletBlanks committed:
Updated README to be clear on how deadline works

ScarletBlanks committed:
Merge branch 'master' of https://github.com/ScarletBlanks/ip

ScarletBlanks committed:
be clearer on teh format of events

ScarletBlanks committed:
update format of readme for deadline

ScarletBlanks committed:
updated README

ScarletBlanks committed:
prepare for release

ScarletBlanks committed:
feat: Updated the files to follow the Java coding standard

ScarletBlanks committed:
Implemented SLAP, added JavaDocs
